Components

memelopez opened the pull request:
##This PR implements: 🆕 💻 💯 

- No linter errors ✔️ 
- Correct use of [Gitflow](https://github.com/microverseinc/linters-config) ✔️ 
- Work is documented in a profressional way ✔️ 
- Deletion of boilerplate CRA ✔️ 
- Creation of `components` directory ✔️ 
- Creation of `Calculator.js` inside `components` ✔️ 
- `Calculator.js` matches the given design ✔️ 
- Does not add any extra design ✔️ 
- Include only UI, no math logic yet ✔️ 
- Use of class-based components ✔️ 
- Calculator is imported in App.js ✔️ 
- Runs in browser with any errors ✔️

Davidosky007 commented on src/components/Calculator.js:
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function Button(props) {
+  const { value } = props;
+  let classes4btn = 'calcBtn';
+  if (value === '0') {
+    classes4btn = 'calcBtn span2';
+  }
+  if (value === '+' || value === 'x' || value === '-' || value === '=') {
+    classes4btn = 'calcBtn orangeBtn';
+  }
```
* Kindly use class-based components for all your components as stated in the project requirement [Check here ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md#:~:text=Use%20class%20based%20components%20(All%20components%20should%20be%20class%2Dbased).)

Davidosky007 reviewed (changes requested):
Hi @memelopez 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well wrritten PR description 👍 
* Nice commit messages


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

memelopez commented:
I changed the Button component to a class-based component although, as I understand from the reading materials of the week, I DID NOT HAVE TO cause I'm only sending it props at the moment. But, I figured I would use state in the future with that component so I changed it to class-based component right now as the reviewer asked for this in required changes. 

The Output component I really don't see any benefit to changing it to class component, it will never use anything more than props. I ask that you don't request me to change it, please. I know how to use class components. 

The change to the ESlint file is to avoid a lint error of useless constructor. When I do something in the constructor further in the project I could take that rule out. 

thank you 👍🏼 😊

MeqdamAlqudah reviewed (approved):
Hi @memelopez , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
